rocAL/rocAL/include/parameters/parameter_random_crop_dec.h

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include "parameter_factory.h"
+#include <thread>
+#include <random>
+
+struct CropWindow {
+  unsigned x, y, H, W;
+  CropWindow() {}
+  CropWindow(unsigned x1, unsigned y1, unsigned h, unsigned w) { x = x1, y=y1, H=h, W=w; }
+  void set_shape(unsigned h, unsigned w) { H=h, W=w; }
+};
+
+typedef std::vector<size_t> Shape;
+
+using AspectRatioRange = std::pair<float, float>;
+using AreaRange = std::pair<float, float>;
+class RocalRandomCropDecParam {
+ public:
+  explicit RocalRandomCropDecParam(
+    AspectRatioRange aspect_ratio_range = { 3.0f/4, 4.0f/3 },
+    AreaRange area_range = { 0.08, 1 },
+    int64_t seed = time(0),
+    int num_attempts_ = 10,
+    int batch_size = 256); // shobi need to set the exact batch size
```

Comment:
This needs to be changed to set exact batch size?

Reply:
No, this has been initialized in image_read_and_decode.cpp with batch_size paramter.